src/components/MozEngineWarning.ts

```diff
@@ -0,0 +1,85 @@
+import { LitElement, html, css } from 'lit'
+
+export class MozEngineWarning extends LitElement {
+  isVisible: boolean = false
+  isLocalAiEnabled: boolean = false
+
+  static properties = {
+    isVisible: { type: Boolean },
+    isLocalAiEnabled: { type: Boolean },
+  }
+
+  constructor() {
+    super()
+  }
+
+  firstUpdated() {
+    this.initLocalStorageData()
+  }
+
+  async initLocalStorageData() {
+    this.isLocalAiEnabled = await browser.permissions.contains({
+      permissions: ['trialML'],
+    })
+    this.isVisible = !this.isLocalAiEnabled
+  }
+
+  handleEnable() {
+    ;(browser.permissions.request as any)({ permissions: ['trialML'] }).then(
+      (granted: boolean) => {
+        this.isVisible = !granted
+      }
+    )
+  }
+
+  static styles = css`
+    :host {
+      --color-bg: #bd2424;
+      --color-fg: #ffffff;
+    }
+
+    .wrapper {
+      background-color: var(--color-bg);
+      color: var(--color-fg);
+      padding: 10px;
+      position: absolute;
+      bottom: 0px;
+    }
+
+    .button {
+      background-color: var(--color-fg);
+      color: var(--color-bg);
+      border: none;
+      padding: 5px 10px;
+      cursor: pointer;
+      font-size: 14px;
+      border-radius: 4px;
+    }
+
+    .actions {
+      display: flex;
+      justify-content: flex-end;
+      margin-top: 10px;
+    }
+  `
+
+  render() {
+    if (!this.isVisible) {
```

Comment:
if we're going to show this at the bottom, why not just add it as a regular instead of optional permission?

https://github.com/Firefox-AI/extension-hub/blob/23b2a4aee7c6243c834e136aa6e212d7ed56c3bf/manifest.json#L21-L22

Reply:
doing a little digging ( and checked that it did not work) It looks like as it currently stands it is only able to be in optional.. https://searchfox.org/mozilla-central/source/toolkit/components/extensions/test/xpcshell/test_ext_permissions_optional_only.js#47
<img width="543" height="428" alt="Screenshot 2025-07-22 at 3 05 37 PM" src="https://github.com/user-attachments/assets/0933739e-57a0-46ac-8d6b-4912aea1f02c" />

Reply:
oh sorry about that, i briefly tried moving it to permissions and got a permission prompt but i think the popup for optional optin confused the flow. indeed seems more complicated